### DISCUSSION TITLE
css-loader: className not directly accessible from the default import when `css-loader` is the only loader and `namedExport: false`

### DISCUSSION THREAD

**Discussion by nyngwang** [webpack/webpack#18081] [Q&A] [answered]
As title. Given that `namedExport: false`,

```css
.someClass { ... }
```

```js
import styles from './test.css'

console.log(styles.locals.someClass) // this will return `className`.
console.log(styles.someClass)        // this will return `undefined`.
```

but when `namedExport: true`,

```js
import styles from './test.css'

// cannot access `className` from the default export at all.
console.log(styles.locals)    // this is undefined.
console.log(styles.someClass) // this is undefined.
```
```js
import * as styles from './test.css'

console.log(styles.someClass) // this will return the `className`.
```

Why are those `className`s not directly accessible from the default export in both cases? But from the comment linked below, it seems that those `className`s are accessible if `style-loader` is also used after `css-loader`. Why this design?

ref: https://github.com/webpack-contrib/css-loader/issues/692#issuecomment-457918645

**Comment by alexander-akait** [comment, marked as answer]
> but when namedExport: true

Because it is a named export, so we don't use `default`, you can use:

```js
import { someClass } from './test.css'
```

> Why are those classNames not directly accessible from the default export in both cases? 

When you have `namedExport: false`, we use the `default` export for CSS code (i.e. CSS code itself and `@import`s), that's is why we need to use `locals` to export classes, we just can't export classes under the `default` export, but when you have `namedExport: true` we use named export to export classes and that is why you can write `import * as styles from './test.css'` or `import { someClass } from './test.css'`

**Reply by nyngwang** [reply]
Thank you cat, I understand it now. It's reasonable that `css-loader` should use `default` for CSS code. I fell asleep.
